### ISSUE TITLE
Release data.table 1.17.0

### ISSUE THREAD

**Issue by MichaelChirico** [Rdatatable/data.table#6616]
We are nearing our planned 6-month cadence for a non-patch release.

Filing this issue to track progress and as a call to flag issues for inclusion in / exclusion from the milestone:

https://github.com/Rdatatable/data.table/milestone/34

Thanks for everyone's work :)

Let's aim for what can be coded up & reviewed by the end of January?

**Comment by TysonStanley** [comment]
That works, will get it live before the 20th deadline too.

**Comment by TysonStanley** [comment]
How are we feeling about meeting the Feb 20 deadline for the whole release? A TON has been done but looks like a few lingering ones. The only one remaining that seems particularly messy is the C API changes. I imagine since we are technically submitting a release in response to a fix on CRAN, they may allow us one more release without having the full C API fixed.

**Comment by MichaelChirico** [comment]
@TysonStanley yes, I think the investment required for the non-API stuff remains large. I suggest trying one more submission with the `NOTES` intact (especially, since a fix requires R 4.5.0 which is not yet officially released, or even scheduled).

I do want to fix up the last translation stuff, #6774 and #6776 have translation contributions that we should merge in, but it requires a bit of time I haven't found yet.

**Comment by TysonStanley** [comment]
Sounds good, we still will ship as `1.17.0` and will communicate with CRAN about the process we are undertaking for the non-API stuff. Let me know when the translation stuff is all complete and will get this off.

**Comment by MichaelChirico** [comment]
@TysonStanley GTG!

**Comment by TysonStanley** [comment]
@MichaelChirico will start the process tonight
